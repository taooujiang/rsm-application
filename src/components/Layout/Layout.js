import React,{Component} from 'react'
import PropTypes from 'prop-types'

export default class Layout extends Component{

  static defaultProps={
    direction:'column',
    position:false
  }
  static propTypes={
    direction:PropTypes.oneOf(['column', 'rows'])
  }

  render(){
    let {children,direction,style,className} = this.props
    let styles=  Object.assign({}, {
        display:'flex',
        flex:'1',
        overflow:'hidden',
        flexDirection:direction
      }, style)
    return (<div className={`layout ${className}`} style={styles}>{children}</div>)
  }
}

class Fixed extends Component{

  static defaultProps={
  }
  static defaultTypes={
  }
  render(){
    let {children,style,width} = this.props
    let styles=  Object.assign({display:'flex',flexDirection: 'column'},style)
    return (<div className="layout-fixed" style={styles}>{children}</div>)
  }
}

class Pane extends Component{
  render(){
    let {children,style} = this.props
    let styles=  Object.assign({}, {display:'flex','flex':'1',position:'relative'}, style)
    return (<div className="layout-pane" style={styles}>{children}</div>)
  }
}

export {Layout,Fixed,Pane}
