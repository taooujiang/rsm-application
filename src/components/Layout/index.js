import Layout,{Fixed,Pane} from './Layout'

export {Fixed,Pane,Layout}
export default Layout
