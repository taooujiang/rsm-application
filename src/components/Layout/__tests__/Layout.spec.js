
import React,{Component} from 'react'
import {Layout,Fixed,Pane} from '../Layout'
import { shallow } from 'enzyme'


describe('测试layout布局组件', function(){
  it('组件render without children',()=>{
    let layoutComponent = shallow(<Layout></Layout>)
    let fixedComponent = shallow(<Fixed></Fixed>)
    let paneComponent = shallow(<Pane></Pane>)
  //  expect(layoutComponent.equals('<div className="layout"></div>').toEqual(true);
  //  expect(fixedComponent.equals('<div className="layout-fixed"></div>').toEqual(true);
  //  expect(paneComponent.equals('<div className="layout-pane"></div>').toEqual(true);
  })
  it('测试组件render带children',()=>{
    let layoutComponent = shallow(<Layout>layout</Layout>)
    expect(layoutComponent.find(".layout").exists()).toBe(true)
    expect(layoutComponent.find(".layout").text()).toEqual("layout")
    let fixedComponent = shallow(<Fixed>fixed</Fixed>)
    expect(fixedComponent.find(".layout-fixed").exists()).toBe(true)
    expect(fixedComponent.find(".layout-fixed").text()).toEqual("fixed")
    let paneComponent = shallow(<Pane>pane</Pane>)
    expect(paneComponent.find(".layout-pane").exists()).toBe(true)
    expect(paneComponent.find(".layout-pane").text()).toEqual("pane")
  })
  it('测试组件props传递',()=>{
    let layoutComponent = shallow(<Layout direction="column"/>)
    expect(layoutComponent.prop('style').flexDirection).toEqual('column')
    expect(layoutComponent).toMatchSnapshot();
  })
  it('测试组件错误props传递',()=>{
    let layoutComponent = shallow(<Layout direction="rows"/>)
    expect(layoutComponent.prop('style').flexDirection).toEqual('rows')
    expect(layoutComponent).toMatchSnapshot();

  })
  it('测试组件nest',()=>{
      let layoutComponent = shallow(<Layout direction="rows">
        <Fixed></Fixed>
        <Pane></Pane>
      </Layout>)
      expect(layoutComponent.contains(<Fixed></Fixed>)).toEqual(true);
      expect(layoutComponent.contains(<Pane></Pane>)).toEqual(true);
  })
  it('测试组件复合nest',()=>{
      let layoutComponent = shallow(<Layout direction="rows">
        <Fixed>header</Fixed>
        <Pane>
          <Layout direction="column">
            <Fixed>side</Fixed>
            <Pane>content</Pane>
          </Layout>
        </Pane>
      </Layout>)
      expect(layoutComponent.contains(<Fixed>header</Fixed>)).toEqual(true);
    //  expect(layoutComponent.find(<Pane/>).childAt(0).type()).toEqual("");
  })
})
