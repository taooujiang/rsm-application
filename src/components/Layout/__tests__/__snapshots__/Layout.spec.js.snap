// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试layout布局组件 测试组件props传递 1`] = `
<div
  className="layout"
  style={
    Object {
      "display": "flex",
      "flex": "1",
      "flexDirection": "column",
      "overflow": "hidden",
    }
  }
/>
`;

exports[`测试layout布局组件 测试组件错误props传递 1`] = `
<div
  className="layout"
  style={
    Object {
      "display": "flex",
      "flex": "1",
      "flexDirection": "rows",
      "overflow": "hidden",
    }
  }
/>
`;
